move-to-sdk/app/io/sphere/sdk/facets/RangeBucketFacetImpl.java

```diff
@@ -0,0 +1,115 @@
+package io.sphere.sdk.facets;
+
+import io.sphere.sdk.search.FacetedSearchExpression;
+import io.sphere.sdk.search.PagedSearchResult;
+import io.sphere.sdk.search.RangeFacetResult;
+import io.sphere.sdk.search.model.*;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class RangeBucketFacetImpl<T> extends BaseFacet<T> implements RangeBucketFacet<T> {
+
+    private final List<FacetRange> selectedRanges;
```

Comment:
Same comment as before: don't forget the type parameters, they are important to keep.